### PR TITLE
Update TestLogicalOperatorsFound

### PR DESCRIPTION
**Pull request by hanneskod**
If I understand the intent of `TestLogicalOperatorsFound` correctly this is how I think it should be written.

The use of `PhpParser\Node\Expr\BinaryOp\NotIdentical` confused me, I don't think I understand what this test did before.  It didn't really work though as it issued a warning when scanning this code (due to the `or` at the end of `$operator`):

```php
if (stristr($line, $operator) !== false) {
    // ...
}
```
With this pr this code pass the test

```php
if (true && true) {
   // ...
}
```

And this code fails the test

```php
if (true and true) {
   // ...
}
```

Is that how the test was intended?

(Related to issue #27)

### CODE REVIEW

File: src/Psecio/Parse/Tests/TestLogicalOperatorsFound.php [hunk 1]
```diff
@@ -5,35 +5,25 @@
...
-            $lines = $file->getLines($attr['startLine']);
-
-            foreach ($lines as $line) {
-                foreach (self::$operators as $operator) {
-                    if (stristr($line, $operator) !== false) {
-                        return false;
-                    }
-                }
-            }
+        if ($node instanceof LogicalAnd || $node instanceof LogicalOr) {
```
Comment (redbeardcreator) [inline src/Psecio/Parse/Tests/TestLogicalOperatorsFound.php]: Can XOR be checked for this way? It's a valid PHP operator, as well.

Comment (hanneskod) [inline src/Psecio/Parse/Tests/TestLogicalOperatorsFound.php]: Yes it is. But xor has no equivalent to || or &&. I am not sure what best practice is here. Do you have a link to a resource discussing this?

**Comment by redbeardcreator** [comment]
No, I don't have a resource on it. However, given that `xor` has lower precedence than the assignment operator, it will behave unexpectedly. So `$x = $a xor $b;` will result in the same result as `$x = $a;` regardless of the value of $b. So I would say it's not safe to use.

Maybe that can be amended and to say it's not safe to use if it's not being used inside a parenthetical expression. But I don't know how easy that would be to test for. I know it can be done by hooking into traversal and maintaining state, but that would be ugly. Works for tracking options (see #12), but would be clunky for this.

**Comment by redbeardcreator** [comment]
I've never actually seen XOR in the wild. Doesn't mean no one has used it, of course,

How does this sound: test for XOR here. If it becomes an issue, move the XOR test to its own test so it can be disabled independently of the AND/OR tests, which have replacements.

**Comment by redbeardcreator** [comment]
BTW, I agree that should write our own parser. That's just bad news.

**Comment by hanneskod** [comment]
Yes let's leave it and see if it becomes an issue..

**Comment by hanneskod** [comment]
This is also ready to merge. Simple tests added..

**Comment by redbeardcreator** [comment]
Looks good to me.

:+1:

**Pull request merged**